creator-node/src/middlewares.js

```diff
@@ -155,52 +165,111 @@ async function getCreatorNodeEndpoints (req, wallet) {
   // Poll discprov until it has indexed provided blocknumber to ensure up-to-date user data.
   let user = null
   const { blockNumber } = req.body
+
   if (blockNumber) {
-    let discprovBlockNumber = -1
     const start2 = Date.now()
 
     // In total, will try for 200 seconds.
     const MaxRetries = 40
     const RetryTimeout = 5000 // 5 seconds
+
+    // Initial delay before polling
     await utils.timeout(1000)
+
+    let discprovBlockNumber = -1
     for (let retry = 1; retry <= MaxRetries; retry++) {
       req.logger.info(`getCreatorNodeEndpoints retry #${retry}/${MaxRetries} || time from start: ${Date.now() - start2} discprovBlockNumber ${discprovBlockNumber} || blockNumber ${blockNumber}`)
+
       try {
         const fetchedUser = await libs.User.getUsers(1, 0, null, wallet)
+
         if (!fetchedUser || fetchedUser.length === 0 || !fetchedUser[0].hasOwnProperty('blocknumber') || !fetchedUser[0].hasOwnProperty('track_blocknumber')) {
           throw new Error('Missing or malformatted user fetched from discprov.')
         }
+
         user = fetchedUser
         discprovBlockNumber = Math.max(user[0].blocknumber, user[0].track_blocknumber)
+
         if (discprovBlockNumber >= blockNumber) {
           break
         }
       } catch (e) { // Ignore all errors until MaxRetries exceeded.
         req.logger.info(e)
       }
+
       await utils.timeout(RetryTimeout)
       req.logger.info(`getCreatorNodeEndpoints AFTER TIMEOUT retry #${retry}/${MaxRetries} || time from start: ${Date.now() - start2} discprovBlockNumber ${discprovBlockNumber} || blockNumber ${blockNumber}`)
     }
 
+    // Error if discprov doesn't return any user for wallet
+    if (!user) {
+      throw new Error(`Failed to retrieve user from discprov after ${MaxRetries} retries. Aborting.`)
+    }
+
+    // Error if discprov has still not indexed to target blockNumber
     if (discprovBlockNumber < blockNumber) {
       throw new Error(`Discprov still outdated after ${MaxRetries}. Discprov blocknumber ${discprovBlockNumber} requested blocknumber ${blockNumber}`)
     }
+  } else if (ensurePrimary && ownEndpoint) {
```

Comment:
is blockNumber taken into account in the `ensurePrimary && ownEndpoint` code block?  because it seems to me that the call to `/audius_users/metadata`  when calling `ensurePrimaryMiddleware` will pass `req.body.blockNumber` if invoked but im not sure that this codeBlock actually gets executed?

Reply:
since this is an `else if` condition with the first `if` being `if(blockNumber)`

Reply:
i don't think blocknumber would be considered here in the else if. this is the fallback in case blocknumber isn't available

Reply:
not sure i follow - if blocknumber is passed the `if (blockNumber)` will be executed, which is correct and `else if (ensurePrimary && ownEndpoint)` should not be executed. maybe easier to chat on call quickly

Reply:
i see, clarified offline - this is just a better fallback function in the cases where an old client makes an outgoing request w/no blocknumber

Reply:
resolving per offline